misc/explain.ninja

```diff
@@ -0,0 +1,23 @@
+# Use like:
```

Comment:
It is not clear how to use this file by reading this comment. I know that all the details are in the conversation for this PR, but this is not enough. Please add a Python3 regression test under `misc/explanation_test.py` that actually invokes all the steps described in the conversation for this PR, and checks that everything works correctly. See `misc/output.py` for inspiration, or add something to it instead.

In all cases, every new feature should have proper unit and regression tests, otherwise maintenance of the code base is complicated (how do we know if a change actually breaks stuff people rely on), and slows down future reviews (there is only so much that human review can catch).